### ISSUE TITLE
在正式执行executeQuery之前调用close方法会导致jvm crash，建议提示异常信息

### ISSUE THREAD

**Issue by shawn-happy** [4paradigm/OpenMLDB#178]
测试代码如下：
```
@Test
  public void test_close_crash() throws Exception {
    SdkOption option = new SdkOption();
    option.setZkCluster(ZK_ENDPOINT);
    option.setZkPath(ZK_PATH);
    SqlExecutor executor = new SqlClusterExecutor(option);
    CallablePreparedStatement callablePreparedStmt =
        executor.getCallablePreparedStmt(DATABASE_NAME, PROCEDURE_NAME);
    setRequestData(callablePreparedStmt, System.currentTimeMillis());
    QueryFuture future = null;
    ResultSet resultSet = null;
    // 在正式执行executeQuery之前调用close方法会导致jvm crash
    executor.close();
    try {
      future = callablePreparedStmt.executeQueryAsync(100, TimeUnit.MILLISECONDS);
      resultSet = future.get();
      List<Map<String, Object>> data = getData(resultSet);
      assertFalse(data.isEmpty());
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }
```

**Comment by dl239** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [bug-denglong](https://trello.com/c/sdJl9Vdl/2-bug-denglong)

**Comment by vagetablechicken** [comment]
router in statement needs fix, must add a ut for it.

**Comment by dl239** [comment]
fixed in #1191
